### PR TITLE
Expose more error details

### PR DESCRIPTION
**Pull request by mahilleb-msft**

### CODE REVIEW

File: adapters/socketio_berkeley.c [hunk 1]
```diff
@@ -679,7 +685,13 @@ int socketio_open(CONCRETE_IO_HANDLE socket_io, ON_IO_OPEN_COMPLETE on_io_open_c
                                 
                                 if (retval != 1)
                                 {
-                                    LogError("Failure: select failure.");
+                                    LogError("Failure: select failure, retval %d, errno %d.", retval, select_errno);
+                                    if (retval == 0) {
+                                        // timeout
+                                        open_result_detailed.code = 9999;
```
Comment (wolfma61) [inline adapters/socketio_berkeley.c:691]: 9999? why not 666? #Resolved

File: adapters/socketio_win32.c [hunk 2]
```diff
@@ -249,18 +249,20 @@ int socketio_open(CONCRETE_IO_HANDLE socket_io, ON_IO_OPEN_COMPLETE on_io_open_c
...
-        result = __FAILURE__;
+        result = open_result_detailed.code = __FAILURE__;
     }
     else
     {
         if (socket_io_instance->io_state != IO_STATE_CLOSED)
         {
             LogError("Failure: socket state is not closed.");
-            result = __FAILURE__;
+            result = open_result_detailed.code = __FAILURE__;
```
Comment (wolfma61) [inline adapters/socketio_win32.c:265]: >__FAILURE__ [](start = 49, length = 11)

shouldn't we report the STATE_CLOSED? #ByDesign

File: adapters/socketio_berkeley.c (hunk 1 above)
Comment (mahilleb-msft) [reply adapters/socketio_berkeley.c:691]: 668 ? neighbor of the beast? #Resolved

File: inc/azure_c_shared_utility/xio.h [hunk 3]
```diff
@@ -33,9 +33,15 @@ DEFINE_ENUM(IO_SEND_RESULT, IO_SEND_RESULT_VALUES);
 
 DEFINE_ENUM(IO_OPEN_RESULT, IO_OPEN_RESULT_VALUES);
 
+// TODO on low-end targets, allow to exclude fields. This struct is passed by value.
```
Comment (wolfma61) [inline inc/azure_c_shared_utility/xio.h]: >// TODO on low-end targe [](start = 0, length = 24)

for this size already? #Resolved

**Review by wolfma61 — APPROVED** [review]
:shipit:

File: adapters/socketio_win32.c (hunk 2 above)
Comment (mahilleb-msft) [reply adapters/socketio_win32.c:265]: we will report IO_OPEN_ERROR with the line number ... it's not really standardized, but we want to be able to identify when we see it. It's more work to really make it meaningful errors codes.

---
In reply to: [212370639](https://github.com/fmegen/azure-c-shared-utility/pull/4#discussion_r212370639) [](ancestors = 212370639)

File: inc/azure_c_shared_utility/xio.h (hunk 3 above)
Comment (mahilleb-msft) [reply inc/azure_c_shared_utility/xio.h]: yeah, i can drop this.

---
In reply to: [212372322](https://github.com/fmegen/azure-c-shared-utility/pull/4#discussion_r212372322) [](ancestors = 212372322)

**Pull request merged**